### DISCUSSION TITLE
coc.nvim co-exist with nvChad?

### DISCUSSION THREAD

**Discussion by xarthurx** [NvChad/NvChad#2826] [Q&A]
Firstly, I know that nvChad is using native lsp and has setup the whole system around it (lspconfig, mason, etc).

However, when coming to specific languages, especially typescript, vuejs, rust (the ones that I can my colleagues use), the native lsp system is really not there yet. You need put quite a lot of effort to make several plugins work together and still have problems from time to time.

I’ve followed and searched for different options on the internet for quite a while, and almost all experienced Vim users for such languages in their working env finally switched back to “coc.vim” (yes. Many of them tried to abandon coc.vim at the beginning) or just switch to VsCode…

That said, for some languages, the “coc.vim” is still the go-to setup.

I’ve used nvChad for a while, and really like it. Question now is, is it possible to install coc.vim so that it co-exist with the native lsp setup? Or how should I setup it?

**Comment by siduck** [comment]
i dont think it'll work welll with cmp etc, only thing i can think of is loading  cmp/lspconfig at filetypes which are not typescript and coc at filetypes like vue ts etc 

Also ask  on r/neovim.     Been   using  neovim for years , for react/svelte/solid  here and never faced any issues

**Comment by xarthurx** [comment]
> i dont think it'll work welll with cmp etc, only thing i can think of is loading cmp/lspconfig at filetypes which are not typescript and coc at filetypes like vue ts etc
> 

Yeah, I know it works fine for react. `Vuejs` has some special issues that is a bit different. And `rust-analyzer` is another one that native lsp doesn't work well so far. native-lsp also have some issues for python, but it is bearable.

So you mean I can have both and each for different languages? For instance, `coc.nvim` only for a few languages and the rest in native-lsp?

**Reply by siduck** [reply]
idk much , but ik it will be a disaster,  plz seek help for it in r/neovim or its matrix or lspconfig repo, there will be experienced devs who'd be able to help you.

Only troublesome lsps ive heard for nvim are from kotlin, java
